index.js

```diff
@@ -824,3 +817,12 @@ function preappend (blocks) {
     this.encryption.encrypt(offset + i, blocks[i], fork)
   }
 }
+
+function ensureEncryption (core, opts) {
+  if (!opts.encryptionKey) return
+  // Only override the block encryption if its either not already set or if
+  // the caller provided a different key.
+  if (!core.encryption || !b4a.equals(core.encryption.key, opts.encryptionKey)) {
```

Comment:
@kasperisager is there actually ever a case where core.encryption is not set here? this is a freshly minted session no?

Reply:
The only case should be if the root session is unencrypted, no?

Reply:
but core is always the session isn't it?

Reply:
the core variable here

Reply:
Yep, and only the root session initializes block encryption. From there it's passed to other sessions. If the root session is unencrypted, `session.encryption` will be `null`, in this case `core.encryption`.

Reply:
Anything I need to do here?